content/news/posts/2024/PSDevCon 2024-PrestaShop-UIKit.md

```diff
@@ -0,0 +1,206 @@
+
+Chromatic interface: 
+
+![chromatic_interface](/assets/images/2024/12/chromaticinterface.png)
+
+### Structure of a component
+
+Creating components with PUIK is simplified thanks to the Hygen template generator, allowing the creation of all files with a single command line! 
+
+Example - creating the awesome-component:
+
+```
+pnpm component awesome-component
+```
+
+From the root folder _templates, Hygen generates all the necessary files for the component or injects new lines of code as needed if existing files need to be updated.
+
+Example of a Hygen file (EJS template language) with configuration options (file path, position of code lines to inject into existing files, etc.):  
+
+![hygen_configuration](/assets/images/2024/12/hygenconfiguration.png)
+
+Review of main files
+
+In the component's src folder, there is an SFC file in Vue 3 (Composition API, using the script setup with the TS language, and a style tag configured with the SCSS language) and a TypeScript file primarily for defining props and emits.
+
+Example - PuikBadge component and its Vue file:
+
+![puik_vue_file](/assets/images/2024/12/puikvuefile.png)
+
+Example - PuikBadge component and its TypeScript file:
+
+![puik_tsfile](/assets/images/2024/12/puiktsfile.png)
+
+Two other subfolders (test and stories) contain, respectively, the Vitest file (based on Jest) for unit testing and the Storybook documentation files. 
+
+Example - draft of awesome-component.spec.ts generated by Hygen with an initial test to verify that the component is indeed an instance of Vue. 
+
+![instance_vue](/assets/images/2024/12/instancevue.png)
+
+Example - draft of awesome-component.storie.ts generated by Hygen for Storybook documentation: 
+
+![storybook_documentation](/assets/images/2024/12/storybookdocumentation.png)
+
+**Puik-theme package and style conventions**
+
+The classes are prefixed with puik- and use Tailwind directives (@apply, @extend) within the SCSS classes. 
+
+Thus, Tailwind classes are never used directly in the Vue template. 
+Example - SCSS file badge.scss:
+
+![scss_file](/assets/images/2024/12/scssfile.png)
+
+**Puik-web-component package**
+
+Web-components are created from Vue files using the native defineCustomElement method. This approach allows converting Vue components into web-components without having to fully recreate a dedicated variant.
+
+The main advantage of this method is that it allows retaining all the existing logic and code of the Vue components, thus making it easier to reuse them in different contexts, including those where using Vue is not possible or desired. 
+
+For more details, refer to the official documentation [here](https://vuejs.org/guide/extras/web-components).
+
+However, there are some limitations. For example, certain advanced features of Vue, such as slotProps, are not available when used as web-components. This may require adjustments or compromises when implementing complex components.
```

Comment:
```suggestion
However, there are some limitations. For example, certain advanced features of Vue, such as `slotProps`, are not available when used as web-components. This may require adjustments or compromises when implementing complex components.
```